src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/UploadContestedCaseDocumentController.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.controllers;
+
+import com.google.common.collect.ImmutableList;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.UploadContestedCaseDocumentsService;
+
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.AUTHORIZATION_HEADER;
+
+@Slf4j
+@RestController
+@RequestMapping(value = "/case-orchestration")
+public class UploadContestedCaseDocumentController implements BaseController {
+
+    @Autowired
+    UploadContestedCaseDocumentsService service;
+
+    @PostMapping(path = "/upload-contested-case-documents", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Handles update Contested Case details and cleans up the data fields based on the options chosen for Contested Cases")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Callback was processed successfully or in case of an error message is attached to the case",
+            response = AboutToStartOrSubmitCallbackResponse.class),
+        @ApiResponse(code = 400, message = "Bad Request"),
+        @ApiResponse(code = 500, message = "Internal Server Error")})
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> uploadCaseDocuments(
+        @RequestHeader(value = AUTHORIZATION_HEADER, required = false) String authToken,
+        @RequestBody CallbackRequest ccdRequest) throws Exception {
+
+        validateCaseData(ccdRequest);
+
+        CaseDetails caseDetails = ccdRequest.getCaseDetails();
+        Map<String, Object> caseData = caseDetails.getData();
+        log.info("Received request to upload Contested case documents for Case ID: {}", caseDetails.getId());
+
+        caseData = service.filterDocumentsToRelevantParty(caseData);
+
+        //service.cleanupUploadCollection(caseData);
```

Comment:
?

Reply:
cleanup of the uploaded document is now done via streams, can remove this in the controller (I thought I had!)